course_info/static/course_info/js/controllers/PeopleController.js

```diff
@@ -14,110 +14,157 @@
+                .then(handlePostSuccess, handlePostError)
+                .finally($scope.updateProgressBar);
+        };
+        $scope.addPeopleToCourse = function(searchTermList) {
 
-                        data.results[0].searchTerm = searchTerm;
-                        data.results[0].action = 'added to';
-                        $scope.clearMessages();
-                        // if there was a partial error, specifically if there was
-                        // an error adding the user to the Canvas course (we caught
-                        // a Canvas API error). The user has been added to the
-                        // coursemanager db, but could not be added to Canvas.
-                        // In this case it's possible that the user will be addded
-                        // during the next Canvas sync. We let the user know about
-                        // the partial failure and that it may correct itself.
-                        if($scope.partialFailureData){
-                            data.results[0].partialFailureData = $scope.partialFailureData
-                        }
+            /* looks up HUIDs, XIDs, and/or email addresses from searchTerms
+             and attempts to add people to the course who do not already have an
+             enrollment.
+             */
 
-                        $scope.success = data.results[0];
-                        $scope.dtInstance.reloadData();
-                    })
-                    .error(function(data, status, headers, config, statusText) {
-                        // log it, then display a warning
-                        $scope.handleAjaxError(data, status, headers, config,
-                                statusText);
-                        $scope.clearMessages();
-                        $scope.addPartialFailure = {
-                            searchTerm: searchTerm,
-                            text: 'Add to course seemed to succeed, but ' +
-                                'we received an error trying to retrieve ' +
-                                "the user's course details.",
-                        };
-                    })
-                    .finally(function(){
-                        $scope.clearSearchResults();
-                        $scope.searchInProgress = false;
+            var membersByUserId = {};
+            $scope.clearMessages();
+            $scope.operationInProgress = true;
+            $scope.tracking.total = searchTermList.length;
+            $scope.updateProgressBar('Looking up ' + $scope.tracking.total
+                + ' people');
+            var memberPromise = $scope.lookupCourseMembers()
+                .then(function updateCourseMembers(members) {
+                    membersByUserId = $scope.getMembersByUserId(members);
+                    return members;
+                }, function courseMemberLookupFailed(memberError) {
+                    $scope.handleAngularDRFError(memberError);
```

Comment:
Annnd the version I checked in is completely missing this method (which was just doing a `$log.error`).  I'll push a fix after the PR.